src/components/cc-global-header.vue

```diff
@@ -96,10 +92,38 @@ export default defineComponent({
   name: "cc-golbal-header",
   beforeCreate() {
     var requestPath = "/wp-json/ccnavigation-header/menu";
-    var requestUrl = this.baseUrl.replace(/\/$/, '') + requestPath;
+    var requestUrl = this.baseUrl.replace(/\/$/, "") + requestPath;
+    var vm = this;
     axios
       .get(requestUrl)
-      .then((response) => (this.menus = response.data));
+      .then((response) => (this.menus = response.data))
+      .catch(function (err) {
```

Comment:
Rather than fail silently on the HTTP request, we should help the user recover from the failed situation. Returning dummy data here may be confusing.

Rather, we can consider having an explicit case for the development environment. For example, the developer can specify `is-development-mode` to render the menu with dummy data, rather than squashing the HTTP error potentially in production. That way, the developer would have an explicit bypass for the HTTP request and we wouldn't need to rely on the NODE_ENV variable, which might not be set to "development" in a downstream integration project like `licenses`.

Reply:
Related docs: https://v3.vuejs.org/guide/component-props.html#passing-a-boolean

Reply:
This is a far better approach. Let me implement

Reply:
with this approach, we'll have to change the hook to `created()` as @TimidRobot earlier suggested.